igc_lib.py

```diff
@@ -71,7 +73,148 @@ def _rawtime_float_to_hms(timef):
+        return distance < self.radius
+        
+class Task:
+    """Stores a task definition and checks if a flight has achieved the turnpoints in the task.
+    Attributes: 
+        turnpoints: A list of Turnpoint objects.
+        start_time: Raw time (seconds past midnight). The time the race starts. The pilots must start at or after this time.
+        end_time: Raw time (seconds past midnight). The time the race ends. The pilots must finish the race at or before this time. 
+                  No credit is given for distance covered after this time.
+    """
+    
+    @staticmethod
+    def create_from_lkt_file(filename):
+        """ Creates Task from LK8000 task file, which is in xml format.
+            LK8000 does not have ESS or task finish time.
+            For the goal, at the moment, Turnpoints can't handle goal cones or lines, for this reason we default to goal_cylinder.
+        """
+                               
+        # Open XML document using minidom parser
+        DOMTree = xml.dom.minidom.parse(filename)
+        task = DOMTree.documentElement
+
+        # Get the taskpoints, waypoints and time gate
+        taskpoints = task.getElementsByTagName("taskpoints")[0]
+        waypoints = task.getElementsByTagName("waypoints")[0]
+        gate = task.getElementsByTagName("time-gate")[0]
+        tpoints = taskpoints.getElementsByTagName("point")
+        wpoints = waypoints.getElementsByTagName("point")
+        start_time = gate.getAttribute("open-time")
+               
+        start_hours, start_minutes = start_time.split(':')
+        start_time = int(start_hours) * 3600 + int(start_minutes) * 60
+        end_time = 86399  #default end_time of 23:59
+        
+        #create a dictionary of names and a list of longitudes and latitudes as the waypoints co-ordinates are stored separate to turnpoint details
+        coords = defaultdict(list)
+        
+        for point in wpoints:    
+            name = point.getAttribute("name")
+            longitude = float(point.getAttribute("longitude"))
+            latitude = float(point.getAttribute("latitude"))
+            coords[name].append(longitude)
+            coords[name].append(latitude)
+
+            
+        # create list of turnpoints    
+        turnpoints = []
+        for point in tpoints:
+            lat = coords[point.getAttribute("name")][1]
+            lon = coords[point.getAttribute("name")][0]
+            radius = float(point.getAttribute("radius"))/1000
+            
+            if point.getAttribute("idx") == "0":
+                if point.getAttribute("Exit") == "true":
+                    kind = "start_exit"
+                else:
+                    kind = "start_enter"
+            else:
+                if point == tpoints[-1]:   # if it is the last turnpoint i.e. the goal
+                    if point.getAttribute("type") == "line":
+                        kind = "goal_cylinder"     # to change once kind 'line' can be processed.
```

Comment:
You can rewrite the comment as `# TODO(kuaka): change to 'line' once we can process it`, TODOs are useful because they easily grab attention once you get used to them :)